File: broadcast_string.rb
#!/usr/bin/env ruby

require 'net/http'
require 'json'

$phrase = "ph'nglui mglw'nafh cthulhu r'lyeh wgah'nagl fhtagn!"
#$phrase = "ei! 000k4k4ei! 22224ei! 24ei!044ei!004222ei!004224222240044"
$token = "53/b8w5nkWTgqhWm00puFJMoBk3NPMMs3TAPAD8eSU0="
$team = 180

uri = URI("https://davar.icfpcontest.org/teams/#{$team}/solutions")
req = Net::HTTP::Post.new uri
req.basic_auth '', $token
req['Content-Type'] = 'application/json'

Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) do |http|
  data = (0..23).map { |i| { problemId: i, seed: 0, tag: "H", solution: $phrase } }
  req.body = data.to_json
  http.request req
end

File: broadcast_string.py
#!/usr/bin/env python

import requests
from requests.auth import HTTPBasicAuth
import json
import sys
import time

TOKEN = "FaMnv73T2LUfm2Q83f3V0DGkslIyAxs8qouqJ5lanQk="
TEAM = 180
URI = "https://davar.icfpcontest.org/teams/{team}/solutions".format(team = TEAM)

SEED = 18
NUM = 24

phrases = [
#	  "ei!"
#	, "ia!"
#	, "r'lyeh"
#	, "yuggoth"
	"deep seven"
	, "chtonians"
#	, "ph'nglui mglw'nafh cthulhu r'lyeh wgah'nagl fhtagn!"
	, "bigboote"
	, "tsathoggua"
	, "unnamable"
	, "yith"
	, "great race of yith"
	, "celeano"
	, "great hall of celeano"
	, "plateau of leng"
	, "abyss"
	, "lost carcosa"
	, "unknown kadath"
	, "the dreamlands"
	, "the underworld"
	, "dreamlands"
	, "underworlds"
]

def test_words(n, seed, words):
	import pdb
	pdb.set_trace()
	for idx, word in enumerate(words):
		resp = requests.post(URI,\
			headers = {'Content-type': 'application/json'},\
			data = json.dumps([{'problemId': n, 'seed': seed, 'tag': "T".format(idx), 'solution': word}]),\
			auth = HTTPBasicAuth('', TOKEN)
		)
		if resp.status_code == 200:
			print("Tag {0} -- OK".format(idx))
		else:
			print("Tag {0} -- Fail".format(idx))
			break
		#time.sleep(300)

if len(sys.argv) != 3:
	print("Usage: {0} seed problem_id".format(sys.argv[0]))
else:
	test_words(sys.argv[2], sys.argv[1], phrases)